ui/Screen/Project/Source.svelte

```diff
@@ -29,18 +29,30 @@
 
   const { id, metadata } = getContext("project");
 
-  $: ({
-    currentPeerId,
-    currentRevision,
-    currentObjectType,
-    currentObjectPath,
-  } = path.parseProjectSourceLocation(
-    $querystring,
-    null,
-    metadata.defaultBranch,
-    ObjectType.Tree,
-    null
-  ));
+  let currentPeerId;
+  let currentRevision;
+  let currentObjectType;
+  let currentObjectPath;
+
+  $: {
+    const parsed = path.parseProjectSourceLocation($querystring);
+
+    currentPeerId = parsed.peerId || null;
+    currentObjectType = parsed.objectType || ObjectType.Tree;
+    currentObjectPath = parsed.objectPath || null;
+
+    if (parsed.revision) {
+      if (JSON.stringify(currentRevision) !== JSON.stringify(parsed.revision)) {
+        currentRevision = parsed.revision;
+      }
```

Comment:
:eyes:

Reply:
Since they are simple objects, maybe better to check the fields manually, or implement a `compare` or `eq` method on the type.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/654/commits/a12ee4e64a06243d566e5522b7bec83f0d021e87